src/App.js

```diff
@@ -1,20 +1,38 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import TodoList  from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
+const useSemiPersistentState = () => {
+
+  
+  const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem('savedTodoList')));
```

Comment:
If your localStorage is empty you will have problems with your todoList value being `null`. Since we are using an array to hold the todos, it would be best to set this up so that it defaults to an empty array.

`JSON.parse(localStorage.getItem('savedTodoList')) || []`
![202212_0850PM-Brave Browser](https://user-images.githubusercontent.com/10133316/208802513-1c8f5191-6bc9-4e6e-a067-eedeb6168cc5.png)

You'll see in the screencap how the TodoList cannot iterate over the null value.